### ISSUE TITLE
Improve 'node' to have some smarts

### ISSUE THREAD

**Issue by iancoleman** [iancoleman/perpetual_auction_currency_game#3]
Add some code that allows nodes to update their bids after the new NB arrives.

Maybe this is a good point to allow various node strategies to be picked. eg random (already in use) but maybe also 'converge toward median' is an easy alternative to implement as well.

Maybe also this is a good time to allow clicking on any individual node to view and change details for that node such as their current bid value or their bidding strategy?

**Comment by happybeing** [comment]
Great to see this game. 

Some things that spring to mind related to node strategies for later, but maybe worth noting now in case it influences design:
- seeding for different starting conditions (number of nodes with each initial strategy)
- node population dynamics (steady, ramp, function, volatility, method of selecting which leave, strategy assigned to those joining)
- sub-populations (groups of nodes with common ownership, owner goal/behaviour and if adaptive, the collusion strategy of the group such as generic algorithm)

Quite a rabbit hole and possibly far more than will be useful or feasible, but maybe there are some nuggets in there?

**Comment by iancoleman** [comment]
Yes there's a lot of room for improvement, thanks for the extra ideas!

One of the early things I want to do is have maybe a 'record' feature or a 'replay' feature or something like that so we can start to share interesting (or even boring) scenarios. It'd be great to be able to start learning something from all this beyond just the individual interaction.

Eventually hooking it up to webrtc or a websocket for group play would be the end goal but probably not going to develop this particular version into that... who knows.

**Comment by iancoleman** [comment]
* https://github.com/iancoleman/perpetual_auction_currency_game/commit/71e43d3cd5cd460c78c54ccd07a3abf83c37a467 - Change node.bid to allow custom calcs of the value
* https://github.com/iancoleman/perpetual_auction_currency_game/commit/593cac9363ea8445f70deddaf7941b6a3598041d - Add docs for setCalcBidFn and afterNbEvent

Still needs UI elements to allow manually viewing and changing node bid calculation function
